Refine LayoutTests/platform/mac-site-isolation/TestExpectations: accessibility

squelart opened the pull request:
#### d745f0855b6ea6e2d4578e35add0f7f083253289
<pre>
Refine LayoutTests/platform/mac-site-isolation/TestExpectations: accessibility
<a href="https://bugs.webkit.org/show_bug.cgi?id=269770">https://bugs.webkit.org/show_bug.cgi?id=269770</a>
<a href="https://rdar.apple.com/123288546">rdar://123288546</a>

Reviewed by Alex Christensen.

* LayoutTests/platform/mac-site-isolation/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/275184@main">https://commits.webkit.org/275184@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eb19c045fdbe1a7f5e187a838779217a76dc99f1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41007 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20020 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43385 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/43567 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37099 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23033 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17351 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/33960 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/41581 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/16942 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35329 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14587 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/14705 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36322 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/44871 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37197 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/36634 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40381 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/15822 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/12980 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/38758 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17441 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/17492 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/5475 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17085 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on LayoutTests/platform/mac-site-isolation/TestExpectations:
```diff
@@ -248,6 +247,8 @@ workers [ Skip ]
 ##########################################################################
 # Tests that crash (or at least crashed when we initially skipped them): #
 ##########################################################################
+accessibility/model-element-attributes.html [ Pass Crash ]
```
Let's skip these like we do the others.  At this point there's no reason to verify that it continues to crash.  If we decide we want to run them to see if any stop crashing, we should switch all of them to [ Pass Crash ] later.  Also, it may go from crashing to failing.

squelart replied:
Makes sense. I'll add a comment to explain the current Skip'ing.

achristensen07 commented on LayoutTests/platform/mac-site-isolation/TestExpectations:
```diff
@@ -393,6 +394,8 @@ imported/w3c/web-platform-tests/html/browsers/the-window-object/named-access-on-
 ###############################################################################
 # Tests that time out (or at least timed out when we initially skipped them): #
 ###############################################################################
+accessibility/aria-menubar-menuitems.html [ Timeout Pass ]
```
Skip.  Having tens of thousands of timeouts will make it take prohibitively long to run.

achristensen07 commented on LayoutTests/platform/mac-site-isolation/TestExpectations:
```diff
@@ -500,6 +503,16 @@ imported/w3c/web-platform-tests/html/browsers/windows/targeting-cross-origin-nes
+accessibility/mac/webkit-scrollarea-position.html [ Pass ImageOnlyFailure ]
+accessibility/model-element-attributes.html [ Pass ImageOnlyFailure ]
+accessibility/progressbar.html [ Pass ImageOnlyFailure ]
+accessibility/tree-update-with-dirty-layout.html [ Pass ImageOnlyFailure ]
```
I'm not sure if it's been done correctly, but the others in this category are [ ImageOnlyFailure Failure ].  Does [ Pass ImageOnlyFailure ] make a test run indicate that there is no issue with these tests?

squelart replied:
Hmm, well spotted!
I copied the results from `run-webkit-tests --iterations=10 ...`. I assumed this meant intermittent failures, so each run would either Pass, or fail with exactly "ImageOnlyFailure".

Indeed, according to https://trac.webkit.org/wiki/TestExpectations, "If multiple expectations are listed, the test is considered "flaky" and any of those results will be considered as expected."
So `[ ImageOnlyFailure Failure ]` should therefore mean a test always fails with either "ImageOnlyFailure" or a plain "Failure".

Focusing on the first test, accessibility/datalist.html, if I run it with `--iterations=10`, I can see it failing about half the time with "ImageOnlyFailure", and it shows the result "accessibility/datalist.html [ ImageOnlyFailure Pass ]", which makes sense to me.

I then tried to run it a few times without `--iterations`, and either it passed (as half-expected), or it failed; but it still reported an unexpected failure with result "accessibility/datalist.html [ ImageOnlyFailure ]" -- I would have expected an *expected* failure there!?

I also tried in mac-site-isolation/TestExpectations: `[ ImageOnlyFailure ]`, and `[ ImageOnlyFailure Failure ]`, but it behaved the same way, still complaining about unexpected failures, and *not* complaining about passes (even though "Pass" is not in the list of expectations).

IIRC you told me that `--site-isolation` runs the test twice (with and without site-isolation), and compares the results? I'll try to have a look at how that works, and why it still complains about results that are in the TestExpectations file, and doesn't complain about passes when "Pass" is not expected...
Please let me know if you have other ideas.

squelart committed:
Refine LayoutTests/platform/mac-site-isolation/TestExpectations: accessibility
https://bugs.webkit.org/show_bug.cgi?id=269770
rdar://123288546

Reviewed by Alex Christensen.

* LayoutTests/platform/mac-site-isolation/TestExpectations:

Canonical link: https://commits.webkit.org/275184@main
